### PR TITLE
dev

### PR DESCRIPTION
**Pull request by AhmedMohsen600**
## What does this PR do
In this PR me and my coding partner check the error about the dev and make the changes.
## Description of the task to be completed
- [x] Display comments feature
- [x] Display popup feature
- [x] Add a new comment feature
- [x] Add a counter for comments
- [x] Add likes and likes
- [x] test comments counter
- [x] test likes counter  

## How should it be tested
Clone the repository
```
git clone https://github.com/lukman155/Rosto.git
```
Navigate to the folder
```
cd Rosto
```
git checkout dev
```
#### Thanks for reviewing my PR 😊

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+  const username = document.querySelector('.name').value;
+  const comment = document.querySelector('.insights').value;
+  const popupId = form.parentElement.dataset.set;
+  await postComment({ item_id: popupId, username, comment });
+
+  const numberOfComments = document.querySelector('.comments').textContent;
+
+  document.querySelector('.comments').textContent = `${
+    Number(numberOfComments) + 1
+  }`;
```
Comment (Santiago220991) [inline src/index.js:188]: I think it is a good idea to add a validation to prevent the user to submit blank entries in your comments. :+1: :smiley: :smiley:

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+  const modal = document.querySelector('.overlay');
+  const img = document.querySelector('.modal-img img');
+  const productTitle = document.querySelector('.product-title');
+  form.parentElement.setAttribute('data-set', id);
+  img.src = item ? item.strMealThumb : placeholderImg;
+  productTitle.textContent = item.strMeal;
+  modal.classList.add('active-modal');
+  commentsCounter(data);
+  displayCommentsList(data);
+};
```
Comment (Santiago220991) [inline src/index.js:92]: Kindly make sure that your project sticks as much as possible to the template provided. 

- I think it is a good idea to add some details about the food in your pop-up windows. :pray: :pray: 

Your design
image

Template design
image

- Kindly make sure that your pop-up windows are being rendered completely. It helps to improve the user's experience.
image
image

File: src/modules/comments.test.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  test('should render comments counter ', () => {
+    document.body.innerHTML = '<p>Comments (<span class="comments"></span>)</p>';
+
+    commentsCounter([
+      { creation_date: '2022-11-24', username: 'jo', comment: 'bo' },
+      { creation_date: '2022-11-24', username: 'mo', comment: 'bo' },
+    ]);
+
+    const commentsText = document.querySelector('.comments').textContent;
+    expect(Number(commentsText)).toEqual(2);
```
Comment (Santiago220991) [inline src/modules/comments.test.js:15]: I think is a good idea to test edge cases. One of them could be to test the counter when you have 0 comments. :+1: :smiley: :smiley:

File: src/modules/counter.test.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+            <div class="card"></div>
+            <div class="card"></div>
+            <div class="card"></div>
+            <div class="card"></div>
+        </div>`;
+    mealCounter();
+  });
+  test('should display the number of meals available', () => {
+    expect(document.querySelector('.meal-count').innerHTML).toBe(
+      '(6 recipes available)',
```
Comment (Santiago220991) [inline src/modules/counter.test.js:19]: Same here I think it is a good idea to test how the counter works if you have 0 meals. :eyes: :eyes:

**Review by Santiago220991 — CHANGES_REQUESTED** [review]
Hi @AhmedMohsen600 @lukman155, 
I'm Santiago your reviewer for this PR. :wave: :wave: 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!



## Suggested changes

_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
